### ISSUE TITLE
sandbox/httpd/patches out of sync with sandbox/httpd/mod_ssl-alpn

### ISSUE THREAD

**Issue by Sp1l** [icing/mod_h2#44]
From `sandbox/httpd/patches/core-protocols.patch`
```
+++ modules/ssl/ssl_private.h	(working copy)
...
+#if OPENSSL_VERSION_NUMBER >= 0x10001000L && !defined(OPENSSL_NO_NEXTPROTONEG) \
+    && !defined(OPENSSL_NO_TLSEXT)
+#define HAVE_TLS_NPN
+#endif
```

From `sandbox/httpd/mod_ssl-alpn/ssl_private.h`
```
/* ALPN Protocol Negotiation */
#if OPENSSL_VERSION_NUMBER >= 0x10002000L && !defined(OPENSSL_NO_TLSEXT)
```

I'm actually more interested to learn what the requirement is than the files being out of sync. Can you please elaborate? I'm building/linking against LibreSSL (which since 2.1.3 has ALPN support again).

**Comment by icing** [comment]
Sorry for any confusion. The mod_ssl-alpn directory is very old and should have been removed long time ago.

The sandbox built downloads the apache httpd 2.4.16 source and patches with core-protocols-release.patch. The NPN define you found in the patch is a leftover from previous changes and I need to clean that up as well. Thanks, good catch.

The change the patch brings to mod_ssl and httpd core is the new `Protocols` directive and the code to negotiate other protocols than `http/1.1` on a connection. mod_ssl provides for the ALPN negotiation. It registers at the Openssl callbacks to trigger the selection and informs httpd core of the outcome.

I am not an expert on LibreSSL. A recent discussion can be found [here](http://apache-http-server.18135.x6.nabble.com/Comparing-LibreSSL-and-OpenSSL-based-on-ApacheTest-t-ssl-results-td5022810.html).

**Comment by icing** [comment]
Fixed in v0.9.6
